### ISSUE TITLE
Add CLI arg for always skipping outputs in dependency block

### ISSUE THREAD

**Issue by blockmar** [gruntwork-io/terragrunt#1078]
Tested with `terragrunt version v0.22.5`

We are seeing issues with running `terragrunt validate` and `terragrunt validate-all` on code that have dependencies to modules that uses non-default providers, in our case to switch to Another IAM role.

(All examples below run with `disable_init = true`)

```
$ cd /path/terraform/something/core-networking
$ tg validate-all
[terragrunt] 2020/03/04 10:12:40 Setting download directory for module /path/terraform/something/core-networking to /path/terraform/something/core-networking/.terragrunt-cache
[terragrunt] 2020/03/04 10:12:40 Setting download directory for module /path/terraform/shared/cgw to /path/terraform/shared/cgw/.terragrunt-cache
[terragrunt] [/path/terraform/something/core-networking] 2020/03/04 10:12:40 Module /path/terraform/something/core-networking must wait for 0 dependencies to finish
[terragrunt] [/path/terraform/something/core-networking] 2020/03/04 10:12:40 Running module /path/terraform/something/core-networking now
[terragrunt] [/path/terraform/something/core-networking] 2020/03/04 10:12:40 Reading Terragrunt config file at /path/terraform/something/core-networking/terragrunt.hcl
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Reading Terragrunt config file at /path/terraform/shared/cgw/terragrunt.hcl
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Running command: terraform --version
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Terraform files in /path/terraform/shared/cgw/.terragrunt-cache/_2DOYr42s4kz2jA6pFdvhSL8Y8o/clPcrJJZUVUWV3Bi0XrCptPqXvM/customer-gateway are up to date. Will not download again.
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Copying files from /path/terraform/shared/cgw into /path/terraform/shared/cgw/.terragrunt-cache/_2DOYr42s4kz2jA6pFdvhSL8Y8o/clPcrJJZUVUWV3Bi0XrCptPqXvM/customer-gateway
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Setting working directory to /path/terraform/shared/cgw/.terragrunt-cache/_2DOYr42s4kz2jA6pFdvhSL8Y8o/clPcrJJZUVUWV3Bi0XrCptPqXvM/customer-gateway
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Detected 1 Hooks
[terragrunt] [/path/terraform/shared/cgw] 2020/03/04 10:12:40 Running command: terraform output -json

Error: The role "arn:aws:iam::1234567890:role/OurOtherIAMRole" cannot be assumed.

  There are a number of possible causes of this - the most common are:
    * The credentials used in order to assume the role are invalid
    * The credentials do not have appropriate permission to assume the role
    * The role ARN is not valid 
```

The `core-networking` code contains a dependency block pointing to `../../shared/cgw`.

No combination of `--terragrunt-working-dir`, `--terragrunt-include-dir`, `--terragrunt-strict-include --terragrunt-ignore-external-dependencies` or `--terragrunt-ignore-dependency-errors` seems to have any effect on terragrunt walking to the `../../shared/cgw` directory.

I also tries setting `skip_output` on the dependency but since the output is needed for normal use I had to set it using a env-variable. But the variable does not seem to work as terragrunt only in some cases skipps running `terraform output` (our goal is to run `terragrunt validate-all` in our root folder)

Maybe if we could provide `--skip-output` globally as a command line option and always use mock-output in validation?

In my opinion either the implementation of the terragrunt command line options are broken or the documentation is incorrect/misleading as I can not find any solution to my problem.

**Comment by yorinasub17** [comment]
The CLI args for ignoring the dependencies for targetting commands of `xxx-all`. However, when you use a `dependency` block, there is no way to resolve that dependency in the parsing logic unless it is configured to skip that. In fact, I would find that surprising if it didn't get the dependency output when you pass in `ignore-external-dependencies`, as that would mean my plan or apply doesn't make sense (since it is working off of mocks).

Having a CLI arg to globally skip outputs makes sense, although I am weary of that leading to surprise, since it only makes sense with `validate-all`, and it only works if you have `mock_outputs` defined. With that said, I think a CLI arg is explicit enough that it should be clear what is happening.

We're buried at the moment and this isn't in our critical path so we are unlikely to get this implemented soon, but a PR to add it is welcome.

**Comment by ryno75** [comment]
I agree that some configurable needs to be added for this situation (running a `validate-all` and having it emulate the `skip_output` behavior (i.e. use `mock_outputs` for dependency) while at the same time _not_ having a `*plan` run skip outputs (else the plan will show projected changes for mock outputs which may reflect a diff in the plan that it should not).  

Another option might be to add a `skip_outputs_allowed_terraform_commands` slice/list var that ensure that skip_outputs only gets applied to certain run types (e.g. `*validate`(

**Comment by bitsofinfo** [comment]
please add, this is completely annoying. I can't initialize providers in anything w/ deps

**Comment by bitsofinfo** [comment]
The only way I can init any of these terragrunt apps is

1. in all dependencies set `skip_outputs = true`
2. in all `inputs` that reference those dependencies `outputs`, comment them out...

Only then can I init anything. 

Is this how its supposed to work or am I missing something. How are others handling this? @ryno75 @blockmar @yorinasub17

**Comment by bitsofinfo** [comment]
some additional info, my project structure is setup following the recommended "best practice" project structure of https://github.com/gruntwork-io/terragrunt-infrastructure-live-example my difference is that I have `dependencies` declared, with `outputs` referenced in `inputs`

**Comment by lobbin** [comment]
> The only way I can init any of these terragrunt apps is
> 
> 1. in all dependencies set `skip_outputs = true`
> 2. in all `inputs` that reference those dependencies `outputs`, comment them out...
> 
> Only then can I init anything.
> 
> Is this how its supposed to work or am I missing something. How are others handling this? @ryno75 @blockmar @yorinasub17

It was a while since I was poking around this, and I'm not working with this anymore, but this is an example of what I have in our terragrunt files:
```
dependency "zone" {
  config_path = "../../../_global/route53/example.com"

  mock_outputs = {
    zone_name = "temporary-zone-name"
  }

  mock_outputs_allowed_terraform_commands = ["validate"]
  skip_outputs                            = tobool(get_env("DISABLE_INIT", "false"))
}

```

And in a main Makefile, the following:
```
validate:
	DISABLE_INIT=true terragrunt validate-all --terragrunt-non-interactive --terragrunt-parallelism 1
```

**Comment by bitsofinfo** [comment]
can something like this be added please?
